app/src/main/java/org/mozilla/tv/firefox/components/locale/LocaleManager.java

```diff
@@ -272,38 +271,38 @@ private void persistLocale(Context context, String localeCode) {
         settings.edit().putString(PREF_LOCALE, localeCode).apply();
     }
 
-    @Nullable
+    @NonNull
     public Locale getCurrentLocale(@NonNull Context context) {
         if (currentLocale != null) {
             return currentLocale;
         }
 
-        final String current = getPersistedLocale(context);
-        if (current == null) {
-            return null;
+        String current = getPersistedLocale(context);
+        if (current != null) {
+            currentLocale = Locales.parseLocaleCode(current);
         }
-        return currentLocale = Locales.parseLocaleCode(current);
-    }
 
-    @NonNull
-    public Boolean currentLanguageIsEnglish(@NonNull Context context) {
-        Locale current = getCurrentLocale(context);
-        // If locale hasn't been updated (i.e., 'current' is null), use system default
-        if (current == null) {
+        if (currentLocale == null) {
             if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
-                current = context.getResources().getConfiguration().getLocales().get(0);
+                currentLocale = context.getResources().getConfiguration().getLocales().get(0);
             } else {
-                current = context.getResources().getConfiguration().locale;
+                currentLocale = context.getResources().getConfiguration().locale;
             }
         }
+
         // In a very small number of cases, this locale will still be null. Most of our
         // userbase uses English as a primary language, so we default to that as a fallback
-        if (current == null) {
+        if (currentLocale == null) {
             Sentry.capture(new AssertionError("Selected locale not available. Falling back to EN"));
-            return true;
+            currentLocale = Locale.US;
         }
-        String language = Locales.getLanguage(current);
-        return language.toLowerCase(current).startsWith("en");
+
+        return currentLocale;
+    }
+
+    @NonNull
+    public Boolean currentLanguageIsEnglish(@NonNull Context context) {
+        return getCurrentLocale(context).getLanguage().equals("en");
```

Comment:
I think we want to be enabling for `en-US` only, not all english locales: I don't think this is right. I sent out an email and we can follow a follow-up if it's wrong.

Reply:
According to the email, we want:
- Pocket in all English locales
- New home channels in en-US only

We'll need to figure that out.

Reply:
Thank you for clearing that up.  Should be an easy fix here.